--- src/dynpmc/Rules.in.link	2011-03-15 02:24:16.000000000 +0100
+++ src/dynpmc/Rules.in	2011-03-15 02:25:38.000000000 +0100
@@ -1,7 +1,7 @@
 # Copyright (C) 2010, Parrot Foundation.
 
 $(DYNEXT_DIR)/dynlexpad$(LOAD_EXT): src/dynpmc/dynlexpad$(O)
-	$(LD)  @ld_out@$(DYNEXT_DIR)/dynlexpad$(LOAD_EXT) src/dynpmc/dynlexpad$(O) $(LINKARGS)
+	$(LD)  @ld_out@$(DYNEXT_DIR)/dynlexpad$(LOAD_EXT) src/dynpmc/dynlexpad$(O) $(LINKARGS) $(ALL_PARROT_LIBS)
 #IF(win32):	if exist $@.manifest mt.exe -nologo -manifest $@.manifest -outputresource:$@;2
 #IF(cygwin or hpux):   $(CHMOD) 0775 $@
 
@@ -18,7 +18,7 @@
 
 
 $(DYNEXT_DIR)/file$(LOAD_EXT): src/dynpmc/file$(O)
-	$(LD)  @ld_out@$(DYNEXT_DIR)/file$(LOAD_EXT) src/dynpmc/file$(O) $(LINKARGS)
+	$(LD)  @ld_out@$(DYNEXT_DIR)/file$(LOAD_EXT) src/dynpmc/file$(O) $(LINKARGS) $(ALL_PARROT_LIBS)
 #IF(win32):	if exist $@.manifest mt.exe -nologo -manifest $@.manifest -outputresource:$@;2
 #IF(cygwin or hpux):   $(CHMOD) 0775 $@
 
@@ -35,7 +35,7 @@
 
 
 $(DYNEXT_DIR)/foo_group$(LOAD_EXT): $(DYNPMC_FOO_OBJS) src/dynpmc/foo_group$(O)
-	$(LD)  @ld_out@$(DYNEXT_DIR)/foo_group$(LOAD_EXT) src/dynpmc/foo_group$(O) $(DYNPMC_FOO_OBJS) $(LINKARGS)
+	$(LD)  @ld_out@$(DYNEXT_DIR)/foo_group$(LOAD_EXT) src/dynpmc/foo_group$(O) $(DYNPMC_FOO_OBJS) $(LINKARGS) $(ALL_PARROT_LIBS)
 #IF(win32):	if exist $@.manifest mt.exe -nologo -manifest $@.manifest -outputresource:$@;2
 #IF(cygwin or hpux):   $(CHMOD) 0775 $@
 
@@ -71,7 +71,7 @@
 
 
 $(DYNEXT_DIR)/gziphandle$(LOAD_EXT): src/dynpmc/gziphandle$(O)
-	$(LD)  @ld_out@$(DYNEXT_DIR)/gziphandle$(LOAD_EXT) src/dynpmc/gziphandle$(O) $(LINKARGS) $(LIB_ZLIB)
+	$(LD)  @ld_out@$(DYNEXT_DIR)/gziphandle$(LOAD_EXT) src/dynpmc/gziphandle$(O) $(LINKARGS) $(LIB_ZLIB) $(ALL_PARROT_LIBS)
 #IF(win32):	if exist $@.manifest mt.exe -nologo -manifest $@.manifest -outputresource:$@;2
 #IF(cygwin or hpux):   $(CHMOD) 0775 $@
 
@@ -93,7 +93,7 @@
 
 
 $(DYNEXT_DIR)/os$(LOAD_EXT): src/dynpmc/os$(O)
-	$(LD)  @ld_out@$(DYNEXT_DIR)/os$(LOAD_EXT) src/dynpmc/os$(O) $(LINKARGS)
+	$(LD)  @ld_out@$(DYNEXT_DIR)/os$(LOAD_EXT) src/dynpmc/os$(O) $(LINKARGS) $(ALL_PARROT_LIBS)
 #IF(win32):	if exist $@.manifest mt.exe -nologo -manifest $@.manifest -outputresource:$@;2
 #IF(cygwin or hpux):   $(CHMOD) 0775 $@
 
@@ -111,7 +111,7 @@
 
 
 $(DYNEXT_DIR)/pccmethod_test$(LOAD_EXT): src/dynpmc/pccmethod_test$(O)
-	$(LD)  @ld_out@$(DYNEXT_DIR)/pccmethod_test$(LOAD_EXT) src/dynpmc/pccmethod_test$(O) $(LINKARGS)
+	$(LD)  @ld_out@$(DYNEXT_DIR)/pccmethod_test$(LOAD_EXT) src/dynpmc/pccmethod_test$(O) $(LINKARGS) $(ALL_PARROT_LIBS)
 #IF(win32):	if exist $@.manifest mt.exe -nologo -manifest $@.manifest -outputresource:$@;2
 
 src/dynpmc/pmc_pccmethod_test.h : src/dynpmc/pccmethod_test.c
@@ -128,7 +128,7 @@
 
 
 $(DYNEXT_DIR)/rotest$(LOAD_EXT): src/dynpmc/rotest$(O)
-	$(LD)  @ld_out@$(DYNEXT_DIR)/rotest$(LOAD_EXT) src/dynpmc/rotest$(O) $(LINKARGS)
+	$(LD)  @ld_out@$(DYNEXT_DIR)/rotest$(LOAD_EXT) src/dynpmc/rotest$(O) $(LINKARGS) $(ALL_PARROT_LIBS)
 #IF(win32):	if exist $@.manifest mt.exe -nologo -manifest $@.manifest -outputresource:$@;2
 #IF(cygwin or hpux):   $(CHMOD) 0775 $@
 
@@ -147,7 +147,7 @@
 
 
 $(DYNEXT_DIR)/rational$(LOAD_EXT): src/dynpmc/rational$(O)
-	$(LD)  @ld_out@$(DYNEXT_DIR)/rational$(LOAD_EXT) src/dynpmc/rational$(O) $(LINKARGS)
+	$(LD)  @ld_out@$(DYNEXT_DIR)/rational$(LOAD_EXT) src/dynpmc/rational$(O) $(LINKARGS) $(ALL_PARROT_LIBS)
 #IF(win32):	if exist $@.manifest mt.exe -nologo -manifest $@.manifest -outputresource:$@;2
 #IF(cygwin or hpux):   $(CHMOD) 0775 $@
 
@@ -165,7 +165,7 @@
 
 
 $(DYNEXT_DIR)/subproxy$(LOAD_EXT): src/dynpmc/subproxy$(O)
-	$(LD)  @ld_out@$(DYNEXT_DIR)/subproxy$(LOAD_EXT) src/dynpmc/subproxy$(O) $(LINKARGS)
+	$(LD)  @ld_out@$(DYNEXT_DIR)/subproxy$(LOAD_EXT) src/dynpmc/subproxy$(O) $(LINKARGS) $(ALL_PARROT_LIBS)
 #IF(win32):	if exist $@.manifest mt.exe -nologo -manifest $@.manifest -outputresource:$@;2
 #IF(cygwin or hpux):   $(CHMOD) 0775 $@
 
--- src/dynoplibs/Rules.in.link	2011-03-15 02:44:15.000000000 +0100
+++ src/dynoplibs/Rules.in	2011-03-15 02:45:36.000000000 +0100
@@ -10,7 +10,7 @@
 #########################
 
 $(DYNEXT_DIR)/obscure_ops$(LOAD_EXT): src/dynoplibs/obscure_ops$(O) $(LIBPARROT)
-	$(LD) @ld_out@$@ src/dynoplibs/obscure_ops$(O) $(LINKARGS)
+	$(LD) @ld_out@$@ src/dynoplibs/obscure_ops$(O) $(LINKARGS) $(ALL_PARROT_LIBS)
 #IF(win32):	if exist $@.manifest mt.exe -nologo -manifest $@.manifest -outputresource:$@;2
 #IF(cygwin or hpux):	$(CHMOD) 0775 $@
 
@@ -25,7 +25,7 @@
 #########################
 
 $(DYNEXT_DIR)/math_ops$(LOAD_EXT): src/dynoplibs/math_ops$(O) $(LIBPARROT)
-	$(LD) @ld_out@$@ src/dynoplibs/math_ops$(O) $(LINKARGS)
+	$(LD) @ld_out@$@ src/dynoplibs/math_ops$(O) $(LINKARGS) $(ALL_PARROT_LIBS)
 #IF(win32):	if exist $@.manifest mt.exe -nologo -manifest $@.manifest -outputresource:$@;2
 #IF(cygwin or hpux):	$(CHMOD) 0775 $@
 
@@ -40,7 +40,7 @@
 #########################
 
 $(DYNEXT_DIR)/trans_ops$(LOAD_EXT): src/dynoplibs/trans_ops$(O) $(LIBPARROT)
-	$(LD) @ld_out@$@ src/dynoplibs/trans_ops$(O) $(LINKARGS)
+	$(LD) @ld_out@$@ src/dynoplibs/trans_ops$(O) $(LINKARGS) $(ALL_PARROT_LIBS)
 #IF(win32):	if exist $@.manifest mt.exe -nologo -manifest $@.manifest -outputresource:$@;2
 #IF(cygwin or hpux):	$(CHMOD) 0775 $@
 
@@ -55,7 +55,7 @@
 #########################
 
 $(DYNEXT_DIR)/bit_ops$(LOAD_EXT): src/dynoplibs/bit_ops$(O) $(LIBPARROT)
-	$(LD) @ld_out@$@ src/dynoplibs/bit_ops$(O) $(LINKARGS)
+	$(LD) @ld_out@$@ src/dynoplibs/bit_ops$(O) $(LINKARGS) $(ALL_PARROT_LIBS)
 #IF(win32):	if exist $@.manifest mt.exe -nologo -manifest $@.manifest -outputresource:$@;2
 #IF(cygwin or hpux):	$(CHMOD) 0775 $@
 
@@ -70,7 +70,7 @@
 #########################
 
 $(DYNEXT_DIR)/debug_ops$(LOAD_EXT): src/dynoplibs/debug_ops$(O) $(LIBPARROT)
-	$(LD) @ld_out@$@ src/dynoplibs/debug_ops$(O) $(LINKARGS)
+	$(LD) @ld_out@$@ src/dynoplibs/debug_ops$(O) $(LINKARGS) $(ALL_PARROT_LIBS)
 #IF(win32):	if exist $@.manifest mt.exe -nologo -manifest $@.manifest -outputresource:$@;2
 #IF(cygwin or hpux):	$(CHMOD) 0775 $@
 
@@ -85,7 +85,7 @@
 #########################
 
 $(DYNEXT_DIR)/sys_ops$(LOAD_EXT): src/dynoplibs/sys_ops$(O) $(LIBPARROT)
-	$(LD) @ld_out@$@ src/dynoplibs/sys_ops$(O) $(LINKARGS)
+	$(LD) @ld_out@$@ src/dynoplibs/sys_ops$(O) $(LINKARGS) $(ALL_PARROT_LIBS)
 #IF(win32):	if exist $@.manifest mt.exe -nologo -manifest $@.manifest -outputresource:$@;2
 #IF(cygwin or hpux):	$(CHMOD) 0775 $@
 
@@ -100,7 +100,7 @@
 #########################
 
 $(DYNEXT_DIR)/io_ops$(LOAD_EXT): src/dynoplibs/io_ops$(O) $(LIBPARROT)
-	$(LD) @ld_out@$@ src/dynoplibs/io_ops$(O) $(LINKARGS)
+	$(LD) @ld_out@$@ src/dynoplibs/io_ops$(O) $(LINKARGS) $(ALL_PARROT_LIBS)
 #IF(win32):	if exist $@.manifest mt.exe -nologo -manifest $@.manifest -outputresource:$@;2
 #IF(cygwin or hpux):	$(CHMOD) 0775 $@
 
--- config/gen/makefiles/root.in.orig	2011-05-17 10:37:50.000000000 +0200
+++ config/gen/makefiles/root.in	2011-06-20 10:02:40.000000000 +0200
@@ -94,7 +94,7 @@
 CC_WARN          = @ccwarn@
 CFLAGS           = $(CC_INC) @ccflags@ @cc_debug@ @clock_best@ $(CC_SHARED)
 LINK_DYNAMIC     = @link_dynamic@
-LINK             = @link@
+LINK             = @link@ $(LDFLAGS)
 LINKFLAGS        = @linkflags@ @link_debug@ @ld_debug@
 LD               = @ld@
 LD_OUT           = @ld_out@

